tensorflow/python/kernel_tests/cwise_ops_test.py

```diff
@@ -55,6 +56,30 @@
 _XOR = lambda x, y: x ^ y
 _INV = lambda x: ~x
 
+# @ operator supported since python 3.5.
+if sys.version_info >= (3, 5):
+  # Use eval() to avoid a compilation error on earlier versions.
+  _MATMUL = lambda x, y: eval("x @ y")
+else:
+  # For earlier versions of python, emulate regular behavior.
+  # Useful to build and test for 3.5+ on earlier versions.
+  def _MATMUL(x, y):
```

Comment:
The matmul operator doesn't really belong in the *componentwise* operator test file.  I think it would be better to make a new test file.  I'll hold off reviewing this file past this point, since I expect you'll have to make a lot of strange workarounds for the fact that matmul is the furthest thing from componentwise.

In particular, you shouldn't need to run tests of `@` if you're not in Python >= 3.5, since that doesn't make any sense.  So `_MATMUL` should be superfluous.